base/applications/taskmgr/graphctl.c

```diff
@@ -12,666 +12,363 @@
-    this->m_dcPlot = 0;
-    this->m_bitmapOldGrid = 0;
-    this->m_bitmapOldPlot = 0;
-    this->m_bitmapGrid = 0;
-    this->m_bitmapPlot = 0;
-    this->m_brushBack = 0;
-
-    this->m_penPlot[0] = 0;
-    this->m_penPlot[1] = 0;
-    this->m_penPlot[2] = 0;
-    this->m_penPlot[3] = 0;
-
-    /* since plotting is based on a LineTo for each new point
-     * we need a starting point (i.e. a "previous" point)
-     * use 0.0 as the default first point.
-     * these are public member variables, and can be changed outside
-     * (after construction).  Therefore m_perviousPosition could be set to
-     * a more appropriate value prior to the first call to SetPosition.
-     */
-    this->m_dPreviousPosition[0] = 0.0;
-    this->m_dPreviousPosition[1] = 0.0;
-    this->m_dPreviousPosition[2] = 0.0;
-    this->m_dPreviousPosition[3] = 0.0;
-
-    /*  public variable for the number of decimal places on the y axis */
-    this->m_nYDecimals = 3;
-
-    /*  set some initial values for the scaling until "SetRange" is called.
-     *  these are protected variables and must be set with SetRange
-     *  in order to ensure that m_dRange is updated accordingly
-     */
-    /*   m_dLowerLimit = -10.0; */
-    /*   m_dUpperLimit =  10.0; */
-    this->m_dLowerLimit = 0.0;
-    this->m_dUpperLimit = 100.0;
-    this->m_dRange      =  this->m_dUpperLimit - this->m_dLowerLimit;  /*  protected member variable */
-
-    /*  m_nShiftPixels determines how much the plot shifts (in terms of pixels)  */
-    /*  with the addition of a new data point */
-    this->m_nShiftPixels     = 4;
-    this->m_nHalfShiftPixels = this->m_nShiftPixels / 2;  /*  protected */
-    this->m_nPlotShiftPixels = this->m_nShiftPixels + this->m_nHalfShiftPixels;  /*  protected */
-
-    /*  Set how big the grid boxes are  */
-    this->m_nGridXPixels = 12;
-    this->m_nGridYPixels = 12;
-
-    /*  Task manager should scroll grid with the graph. This variable contains the  */
-    /*  amount of pixels the grid paint origin should shift to left  */
-    this->m_nGridOffsetPixels = 0;
-
-    /*  background, grid and data colors */
-    /*  these are public variables and can be set directly */
-    this->m_crBackColor = RGB(  0,   0,   0);  /*  see also SetBackgroundColor */
-    this->m_crGridColor = RGB(  0, 128,  64);  /*  see also SetGridColor */
-    this->m_crPlotColor[0] = RGB(255, 255, 255);  /*  see also SetPlotColor */
-    this->m_crPlotColor[1] = RGB(100, 255, 255);  /*  see also SetPlotColor */
-    this->m_crPlotColor[2] = RGB(255, 100, 255);  /*  see also SetPlotColor */
-    this->m_crPlotColor[3] = RGB(255, 255, 100);  /*  see also SetPlotColor */
-
-    /*  protected variables */
-    for (i = 0; i < MAX_PLOTS; i++)
+    HDC     hdc, hdcg;
+    HBITMAP hbmOld;
+    PBYTE   t;
+    UINT    Size;
+    INT     p;
+    RECT    rc;
+
+    inst->hParentWnd = hParentWnd;
+    inst->hWnd = hWnd;
+
+    Size = GetSystemMetrics(SM_CXSCREEN);
+    if (!Size)
+        Size = 1920;
```

Comment:
Arbitrary value. In which case could the GetSystemMetrics fail?

Reply:
it doesn't. but is it the case for ROS GetSystemMetrics? for the case, it fails, that value is a balanced length for the bitmap, holding the history. this could only cause artifacts if some ReactOS user uses 4K monitor AND ROS GetSystemMetrics fails to say that.

Reply:
> but is it the case for ROS GetSystemMetrics?

It doesn't fail more than on Windows, why would it?

Reply:
so, there isn't problem then? it will always succeed, returning the desired width. :)